libs/ui/src/unconfigured_election_screen.tsx

```diff
@@ -7,24 +7,42 @@ import { CenteredLargeProse } from './centered_large_prose';
 
 interface Props {
   usbDriveStatus: UsbDriveStatus;
+  isElectionManagerAuth: boolean;
   backendConfigError?: BallotPackageConfigurationError;
 }
 
 export function UnconfiguredElectionScreen({
   usbDriveStatus,
+  isElectionManagerAuth,
   backendConfigError,
 }: Props): JSX.Element {
   const errorMessage = (() => {
+    // VxScan backend auth requires election config before authorizing pollworkers,
+    // but the frontend is somehow aware of the user type. So we can check auth status
+    // in the frontend and handle it here, but we can't rely on the backend to return a
+    // BallotPackageConfigurationError for this particular case
```

Comment:
Sorry, this comment is a little unclear. 

Without this check, the pollworker still can't authenticate but we don't handle it gracefully. There's an extra screen where we show them the [default error message](https://github.com/votingworks/vxsuite/pull/3184/files/ce0cc7b70c709d969e4b77038ff1e2a9473e44bc#diff-cfd3ccb7190762e234b7952b0c2cab21a3254f4fee8fd657c139d17d51470c92R28-R30):
https://user-images.githubusercontent.com/1060688/227602983-9b6621df-c314-4828-a64c-a58a2e38ede8.mov

The steps here are:
1. Unconfigured machine prompts for election manager card
2. User inserts a pollworker card
3. `UnconfiguredElectionScreen` fails the first check because no USB drive is mounted and returns the error you see
4. User inserts USB with ballot package
5. VxScan returns and handles [this error](https://github.com/votingworks/vxsuite/pull/3184/files/ce0cc7b70c709d969e4b77038ff1e2a9473e44bc#diff-cfd3ccb7190762e234b7952b0c2cab21a3254f4fee8fd657c139d17d51470c92R41)
6. I disconnect the USB and do a successful configuration at the end of the video to show that the machine was not configured by the pollworker even though the USB was inserted

Ideally we'd make a check for the `'user_role_not_allowed'` `backendConfigError` before the check for unmounted USB drive. But as you point out, we never reach that check because the backend refuses to auth with
```
{"source":"vx-scan-backend","eventId":"auth-login","eventType":"user-action","user":"poll_worker","message":"User failed login.","disposition":"failure","reason":"machine_not_configured"}
```

With this check the behavior can be seen in the PR description under "Attempting to auth as a pollworker before election is configured" (I just changed this from "Authing as a pollworker..." since the pollworker isn't actually completing authentication as you point out).

In writing this up I realized this comment is very misleading; we don't know pollworker auth status. Rather, we know there is a lack of authenticated election manager. Since only election managers can configure ballot packages, I believe this check is correct, but the comment should be clearer about that.

There's also a question of whether the backend check for wrong user role is _too_ much redundancy, since both the frontend and auth layer+[this check](https://github.com/votingworks/vxsuite/pull/3184/files/ce0cc7b70c709d969e4b77038ff1e2a9473e44bc#diff-80d3b0dac90c1c3370e2a5dc4fa423885f5fbd2c9ac5745f34e01057a8820fb2R95-R101) prevent it.

Reply:
^ Thanks for writing all this up! Very helpful for discussion. Just documenting that we discussed this offline and decided to get rid of redundant checks, leaving just the checks that can actually be hit by users (and not the checks that could one day be hit because of code regressions - we can instead use tests to prevent regressions)

Reply:
https://github.com/votingworks/vxsuite/pull/3184/commits/af13e5655cf0af6416a5fa50292558f281c2c993 prevents regressions that would render the ballot package screen for system admins logging into an unconfigured machine.